### PR TITLE
[NFC] Simplify `Tuple` by making it an alias of `TypeList`

### PR DESCRIPTION
**Pull request by tlively**
Rather than wrap a `TypeList`, make `Tuple` an alias of `TypeList`. This means
removing `Tuple::toString`, but that had no callers and was of limited use for
debugging anyway. In return, the use of tuples becomes much less verbose.

In the future, it may make sense to remove one of `Tuple` and `TypeList`.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5775** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5775" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5775?utm_source=stack-comment).

File: src/wasm-type.h [hunk 1]
```diff
@@ -453,31 +455,6 @@ class RecGroup {
...
-
-  bool operator==(const Tuple& other) const { return types == other.types; }
-  bool operator!=(const Tuple& other) const { return !(*this == other); }
-  std::string toString() const;
-
-private:
-  void validate() {
-#ifndef NDEBUG
-    for (auto type : types) {
-      assert(type.isSingle());
```
Comment (kripken) [inline src/wasm-type.h:475]: Do we have this logic already in the validator? If not it might be good to add.

**Review by kripken — APPROVED** [review]
lgtm % question

File: src/wasm-type.h (hunk 1 above)
Comment (tlively) [reply src/wasm-type.h:475]: Oh right, I definitely need to replace this with something...

**Pull request merged**